client/my-sites/media-library/list.jsx

```diff
@@ -152,6 +167,15 @@ export const MediaLibraryList = React.createClass( {
 		return 'item-' + item.ID;
 	},
 
+	getGroupLabel: function( dateString ) {
+		const date = new Date( dateString );
+		return `${ months[ date.getMonth() ].slice( 0, 3 ) } ${ date.getDate() }`;
```

Comment:
I think that years could be handy, though I'm not sure if the fit the current design. At least not in the labels we have now.  
I think the best way to handle this would be to introduce proper headers for each year, but given the current implementation that should get its own PR.
I'll make the year visible for all but the current year.